src/main/scala/io/snyk/plugin/ui/state/Navigator.scala

```diff
@@ -56,27 +60,76 @@ object Navigator extends IntellijLogging {
       * Use MavenProjectsManager to open the editor and highlight where the specified artifact
       * is imported.
       */
-    override def navToArtifact(
+    override def navigateToDependency(
       group: String,
       name: String,
       projectId: String
-    ): Future[Unit] = idToProject(projectId) map { mp =>
-      val p = Promise[Unit]
+    ): Future[Unit] = idToProject(projectId) map { buildToolProject =>
+      val promisedUnit = Promise[Unit]
+
       ApplicationManager.getApplication.invokeLater { () =>
         log.info(s"Navigating to Artifact: $group : $name in $projectId")
-        p complete Try {
-          val file = mp.getFile
-          log.debug(s"  file: $file")
-          val artifact = mp.findDependencies(group, name).asScala.head
-          log.debug(s"  artifact: $artifact")
-          val nav = MavenNavigationUtil.createNavigatableForDependency(project, file, artifact)
-          nav.navigate(true)
+
+        promisedUnit complete Try {
+          val projectType = buildToolProject.getType
+
+          if (ProjectType.MAVEN == projectType) {
```

Comment:
not a must at this point, but is there a `switch` in Scala? Seems a more appropriate pattern here

Reply:
Thank you, Yulia! It's a good point.

Reply:
please change it then to `switch` in the next PR :)